### PR TITLE
test: Creating v1 yaml tests from existing v0 tests

### PR DESCRIPTION
**Pull request by johanfylling**
Fixes: #6864

### CODE REVIEW

File: test/cases/testdata/v1/refheads/test-generic-refs.yaml [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+                - b
+                - c
+            b:
+                - a
+                - c
+                - foo
+            c:
+                - a
+                - b
+# FIXME: Compiler NPE
```
Comment (johanfylling) [inline test/cases/testdata/v1/refheads/test-generic-refs.yaml:154]: For tracking purposes, I'll fix this in a separate PR.

File: test/cases/internal/fmtcases/main.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			for i, module := range testCase.Modules {
+				bs, err := format.SourceWithOpts(fmt.Sprintf("mod%d.rego", i), []byte(module),
+					format.Opts{
+						ParserOptions: &ast.ParserOptions{
+							RegoVersion: sourceRegoVersion,
+						},
+						RegoVersion: targetRegoVersion,
+					})
+				if err != nil {
+					fmt.Printf("Error formatting module %s %s:%d: %v\n", path, testCase.Note, i, err)
```
Comment (johanfylling) [inline test/cases/internal/fmtcases/main.go:102]:
```
Error formatting module testdata/v0/all/test-all-0027.yaml all/empty set:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0028.yaml all/empty array:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0029.yaml all/set success:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0030.yaml all/array success:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0031.yaml all/set fail:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0032.yaml all/array fail:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/all/test-all-0033.yaml all/other types:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: all
Error formatting module testdata/v0/any/test-any-0034.yaml any/empty set:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0035.yaml any/empty array:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0036.yaml any/set success:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0037.yaml any/array success:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0038.yaml any/set fail:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0039.yaml any/array fail:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/any/test-any-0040.yaml any/other types:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: any
Error formatting module testdata/v0/casts/test-casts-0077.yaml casts/null valid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_null
Error formatting module testdata/v0/casts/test-casts-0078.yaml casts/null invalid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_null
Error formatting module testdata/v0/casts/test-casts-0079.yaml casts/string invalid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_string
Error formatting module testdata/v0/casts/test-casts-0080.yaml casts/boolean valid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_boolean
Error formatting module testdata/v0/casts/test-casts-0081.yaml casts/boolean invalid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_boolean
Error formatting module testdata/v0/casts/test-casts-0082.yaml casts/obj valid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_object
Error formatting module testdata/v0/casts/test-casts-0083.yaml casts/obj invalid:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_object
Error formatting module testdata/v0/netcidroverlap/test-netcidroverlap-0084.yaml netcidroverlap/cidr match:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: net.cidr_overlap
Error formatting module testdata/v0/netcidroverlap/test-netcidroverlap-0085.yaml netcidroverlap/cidr mismatch:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: net.cidr_overlap
Error formatting module testdata/v0/regexmatch/test-regexmatch-0855.yaml regexmatch/re_match:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/regexmatch/test-regexmatch-0856.yaml regexmatch/re_match: undefined:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/regexmatch/test-regexmatch-0857.yaml regexmatch/re_match: bad pattern err:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/regexmatch/test-regexmatch-0858.yaml regexmatch/re_match: ref:0: 1 error occurred: mod0.rego:5: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/regexmatch/test-regexmatch-0859.yaml regexmatch/re_match: raw:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/regexmatch/test-regexmatch-0860.yaml regexmatch/re_match: raw: undefined:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: re_match
Error formatting module testdata/v0/sets/test-sets-0872.yaml sets/set_diff: refs:0: 1 error occurred: mod0.rego:10: rego_type_error: deprecated built-in function calls in expression: set_diff
Error formatting module testdata/v0/toarray/test-toarray-0071.yaml toarray/array input:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_array
Error formatting module testdata/v0/toarray/test-toarray-0072.yaml toarray/set input:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_array
Error formatting module testdata/v0/toarray/test-toarray-0073.yaml toarray/bad type:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_array
Error formatting module testdata/v0/toset/test-toset-0074.yaml toset/array input:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_set
Error formatting module testdata/v0/toset/test-toset-0075.yaml toset/set input:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_set
Error formatting module testdata/v0/toset/test-toset-0076.yaml toset/bad type:0: 1 error occurred: mod0.rego:4: rego_type_error: deprecated built-in function calls in expression: cast_set
```

File: test/cases/testdata/v1/refheads/test-generic-refs.yaml (hunk 1 above)
Comment (johanfylling) [reply test/cases/testdata/v1/refheads/test-generic-refs.yaml]: https://github.com/open-policy-agent/opa/pull/6931

File: test/cases/internal/fmtcases/main.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/format"
+	"github.com/open-policy-agent/opa/test/cases"
+	"github.com/open-policy-agent/opa/util"
+	"gopkg.in/yaml.v3"
+)
+
+func main() {
+	if len(os.Args) < 5 {
+		fmt.Println("Usage: main <source> <source-version> <target> <target-version>")
```
Comment (ashutosh-narkar) [inline test/cases/internal/fmtcases/main.go:23]: Nit: Maybe `source-dir` and `target-dir` for clarification

File: test/cases/internal/fmtcases/main.go (hunk 2 above)
Comment (ashutosh-narkar) [reply test/cases/internal/fmtcases/main.go:102]: These ones will need manual intervention, correct?

**Review by ashutosh-narkar — COMMENTED** [review]
This looks good!

File: test/cases/internal/fmtcases/main.go (hunk 2 above)
Comment (johanfylling) [reply test/cases/internal/fmtcases/main.go:102]: Correct. They've been fixed in this PR. Most was just exercising deprecated built-ins, so them I simply removed.

**Review by ashutosh-narkar — APPROVED** [review]
LGTM

**Pull request merged**